addons/utm/models/utm_source.py

```diff
@@ -72,12 +72,17 @@ def create(self, vals_list):
         return super().create(vals_list)
 
     def write(self, values):
-        if values.get(self._rec_name) and not values.get('name'):
-            values['name'] = self.env['utm.source']._generate_name(self, values[self._rec_name])
-        if values.get('name'):
-            values['name'] = self.env['utm.mixin']._get_unique_names(self._name, [values['name']])[0]
-
-        super().write(values)
+        self.ensure_one()
+        name = values.get('name')
+        if values.get(self._rec_name) and not name:
+            name = self.env['utm.source']._generate_name(self, values[self._rec_name])
+        if name == self.name:
+            super().write(values)
```

Comment:
A potential issue is that if the name doesn't perfectly adhere to the result of `_generate_name`, the record will again get a number when it's not necessary.

Not a massive issue but would be good if it can be avoided somehow

Reply:
@reth-odoo can u give an example? i don't fully understand what u mean. if they aren't equal it supposed to check  if the new name will be unique or not (cuz it might be the same as the one 1 year ago in a past), if it's unique then we are good, the name will stay the same, but if there existed source with similar name, then it should get [1] at the end

Reply:
`rec.name == 'name '`
`self.env['utm.source']._generate_name(self, 'name ') == 'name'`
`'name ' != 'name'`
-> 'name ' becomes 'name [2]' because of the way `generate unique name` is implemented (notably, it ignores blank spaces)

But maybe this version is better for other cases actually, I'm not sure :/